HEC.FDA.ModelTest/integrationtests/UncertainStageDamageTests.cs

```diff
@@ -0,0 +1,289 @@
+        private static PointM pointM = new PointM();
+
+        //First impact area structures
+        private static int impactAreaID_34 = 34;
+        private static Structure structure1 = new Structure(fid: 1, point: pointM, firstFloorElevation: 473, val_struct: 100, st_damcat: residentialDamCat, occtype: res2OccType, impactAreaID: impactAreaID_34);
+        private static Structure structure2 = new Structure(fid: 2, point: pointM, firstFloorElevation: 474, val_struct: 200, st_damcat: residentialDamCat, occtype: res2OccType, impactAreaID: impactAreaID_34);
+        private static Structure structure3 = new Structure(fid: 3, point: pointM, firstFloorElevation: 473.5, val_struct: 300, st_damcat: residentialDamCat, occtype: res1OccType, impactAreaID: impactAreaID_34);
+        private static Structure structure4 = new Structure(fid: 4, point: pointM, firstFloorElevation: 474.5, val_struct: 400, st_damcat: residentialDamCat, occtype: res1OccType, impactAreaID: impactAreaID_34);
+
+        //Second impact area structures 
+        private static int impactAreaID_56 = 56;
+        private static Structure structure5 = new Structure(fid: 5, point: pointM, firstFloorElevation: 473, val_struct: 100, st_damcat: residentialDamCat, occtype: res2OccType, impactAreaID: impactAreaID_56);
+        private static Structure structure6 = new Structure(fid: 6, point: pointM, firstFloorElevation: 474, val_struct: 200, st_damcat: residentialDamCat, occtype: res2OccType, impactAreaID: impactAreaID_56);
+        private static Structure structure7 = new Structure(fid: 7, point: pointM, firstFloorElevation: 473.5, val_struct: 300, st_damcat: residentialDamCat, occtype: res1OccType, impactAreaID: impactAreaID_56);
+        private static Structure structure8 = new Structure(fid: 8, point: pointM, firstFloorElevation: 474.5, val_struct: 400, st_damcat: residentialDamCat, occtype: res1OccType, impactAreaID: impactAreaID_56);
+
+        private static StructureSelectionMapping map = new StructureSelectionMapping(false, false, null, null, null, null, null, null, null, null, null, null, null, null, null, null);
+
+        private static Inventory impactArea_34_StructureInventory = new Inventory(occupancyTypes, new List<Structure>() { structure1, structure2, structure3, structure4 });
+        private static Inventory impactArea_56_StructureInventory = new Inventory(occupancyTypes, new List<Structure>() { structure5, structure6, structure7, structure8 });
+
+        #endregion
+
+        #region Other objects 
+        private static ConvergenceCriteria convergenceCriteria = new ConvergenceCriteria(minIterations: 100, maxIterations: 200000);
+        RandomProvider randomProvider = new RandomProvider(seed: 1234);
+        #endregion
+
+        #region Expected Values
+        //The modeling used to generate the expected values can be found at the below link. 
+        //https://drive.google.com/file/d/17IvI01PBXa0C97dlBOKLn0Z0odGTJ663/view?usp=share_link
```

Comment:
You'll never get the credit you deserve for doing this. but it's pretty freakin' cool.

Reply:
thanks man